ports/bsd/rs485.c

```diff
@@ -0,0 +1,899 @@
+	// the current baud rate if the IOSSIOSPEED ioctl was used but will instead return the speed set by the last call
+	// to cfsetspeed.
+    
+    printf("Input baud rate changed to %d\n", (int) cfgetispeed(&options));
+    printf("Output baud rate changed to %d\n", (int) cfgetospeed(&options));
+    
+    // Cause the new options to take effect immediately.
+    if (tcsetattr(fileDescriptor, TCSANOW, &options) == -1) {
+        printf("Error setting tty attributes %s - %s(%d).\n",
+               bsdPath, strerror(errno), errno);
+        goto error;
+    }
+    
+    // To set the modem handshake lines, use the following ioctls.
+    // See tty(4) <x-man-page//4/tty> and ioctl(2) <x-man-page//2/ioctl> for details.
+    
+    // Assert Data Terminal Ready (DTR)
+    if (ioctl(fileDescriptor, TIOCSDTR) == -1) {
+        printf("Error asserting DTR %s - %s(%d).\n",
+               bsdPath, strerror(errno), errno);
+    }
+    
+    // Clear Data Terminal Ready (DTR)
+    if (ioctl(fileDescriptor, TIOCCDTR) == -1) {
+        printf("Error clearing DTR %s - %s(%d).\n",
+               bsdPath, strerror(errno), errno);
+    }
+    
+    // Set the modem lines depending on the bits set in handshake
+    handshake = TIOCM_DTR | TIOCM_RTS | TIOCM_CTS | TIOCM_DSR;
+    if (ioctl(fileDescriptor, TIOCMSET, &handshake) == -1) {
+        printf("Error setting handshake lines %s - %s(%d).\n",
+               bsdPath, strerror(errno), errno);
+    }
+    
+    // To read the state of the modem lines, use the following ioctl.
+    // See tty(4) <x-man-page//4/tty> and ioctl(2) <x-man-page//2/ioctl> for details.
+    
+    // Store the state of the modem lines in handshake
+    if (ioctl(fileDescriptor, TIOCMGET, &handshake) == -1) {
+        printf("Error getting handshake lines %s - %s(%d).\n",
+               bsdPath, strerror(errno), errno);
+    }
+    
+    printf("Handshake lines currently set to %d\n", handshake);
+	
+	unsigned long mics = 1UL;
+    
+	// Set the receive latency in microseconds. Serial drivers use this value to determine how often to
+	// dequeue characters received by the hardware. Most applications don't need to set this value: if an
+	// app reads lines of characters, the app can't do anything until the line termination character has been
+	// received anyway. The most common applications which are sensitive to read latency are MIDI and IrDA
+	// applications.
+	
+	if (ioctl(fileDescriptor, IOSSDATALAT, &mics) == -1) {
+		// set latency to 1 microsecond
+        printf("Error setting read latency %s - %s(%d).\n",
+               bsdPath, strerror(errno), errno);
+        goto error;
+	}
+    
+    // Success
+    return fileDescriptor;
+    
+    // Failure path
+error:
+    if (fileDescriptor != -1) {
+        close(fileDescriptor);
+    }
+    
+    return -1;
+}
+
+// Given the file descriptor for a serial device, close that device.
```

Comment:
Although it probably won't matter in this port, the rest of the library uses C comment blocks, not C++, as the library code needs to work with C89 compilers that don't support `//`.